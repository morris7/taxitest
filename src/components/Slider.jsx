import React, { Component } from "react";
import PropTypes from 'prop-types';

class Slider extends Component {

    constructor(props){
        super(props);

        this.state = {
            sliderValue: 10
        }

        this.updateValue = this.updateValue.bind(this);
        this.search = this.search.bind(this);
    }

    render() {
        const mapStyles = {
          position: 'absolute',
          top: '600px'
        }
        const {min, max} = this.props;
        return (
            <form style={mapStyles}>
                <input type="range" min={min} max={max} value={this.state.sliderValue} className="slider" id="carcount"
                    onChange={this.updateValue}/>
                <span>{this.state.sliderValue}</span>
                <button onClick={this.search}>Update search...</button>
            </form>
        );
    }

    updateValue(event, value){
        let { target } = event;

        this.setState({ sliderValue: target.value });
    }

    search(e){
      e.preventDefault();
      this.props.callback(this.state.sliderValue);
    }
}

Slider.propTypes = {
  max: PropTypes.number,
  min: PropTypes.number,
  callback: PropTypes.func
}

export default Slider;
