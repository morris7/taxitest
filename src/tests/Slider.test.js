import React from 'react';
import ReactDOM from 'react-dom';
import Slider from '../components/Slider';
const ReactTestRenderer = require('react-test-renderer');

describe('when rendering a Slider component', () => {
  it('should render without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Slider />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should match snapshot', () => {
    const component = ReactTestRenderer.create(<Slider />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
