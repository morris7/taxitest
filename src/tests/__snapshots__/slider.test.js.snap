// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when rendering a Slider component should match snapshot 1`] = `
<form
  style={
    Object {
      "position": "absolute",
      "top": "600px",
    }
  }
>
  <input
    className="slider"
    id="carcount"
    onChange={[Function]}
    type="range"
    value={10}
  />
  <span>
    10
  </span>
  <button
    onClick={[Function]}
  >
    Update search...
  </button>
</form>
`;
